### PR TITLE
[#5748][refactor]get more reasonable monitoring indexes of the zk cluster

### PR DESCRIPTION
**Pull request by jon-qj**
add warning state on the front-end
each zk node shows monitoring indexes correctly

## Purpose of the pull request

This pull request gets more reasonable monitoring indexes of the cluster

## Brief change log

add warning state on the front-end
each zk node shows monitoring indexes correctly

## Verify this pull request

Manually verified the change by testing locally.

### CODE REVIEW

**Comment by ruanwenjun** [comment]
@jon-qj You may need to submit this pr to dev branch.
https://dolphinscheduler.apache.org/en-us/community/development/submit-code.html

**Comment by CalvinKirs** [comment]
> @jon-qj You may need to submit this pr to dev branch.
> https://dolphinscheduler.apache.org/en-us/community/development/submit-code.html

You can commit to 1.3.7-prepare, the dev branch piece of functionality has been removed ：）

**Comment by jon-qj** [comment]
ok,I’ll try again.Thanks a lot.
 
 
 
 
 
 
 
 
 
 
 

  -- 
 
 
 发自新浪邮箱客户端 
 
 
  
  在 7月8日 11:44，Kirs ***@***.***> 写道： 
   
   
   @jon-qj You may need to submit this pr to dev branch.
 https://dolphinscheduler.apache.org/en-us/community/development/submit-code.html 
   
  You can commit to 1.3.7-prepare, the dev branch piece of functionality has been removed ：）

**Comment by CalvinKirs** [comment]
@ruanwenjun The back end looks fine. please take a look again.

**Comment by ruanwenjun** [comment]
@jon-qj I have test on my local environment, when the zookeeper down it will display the message on the ui, this looks good.
image

But you don't need to submit a new pr #5748, you can edit this pr commit into 1.3.7-prepare
image

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/FourLetterWordMain.java [hunk 1]
```diff
@@ -25,10 +25,7 @@
 import java.io.IOException;
 import java.io.InputStreamReader;
 import java.io.OutputStream;
-import java.net.InetAddress;
-import java.net.InetSocketAddress;
-import java.net.Socket;
-import java.net.SocketTimeoutException;
+import java.net.*;
```
Comment (ruanwenjun) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/FourLetterWordMain.java]: Don't use import *.
https://stackoverflow.com/questions/147454/why-is-using-a-wild-card-with-a-java-import-statement-bad

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/ZooKeeperState.java [hunk 2]
```diff
@@ -43,6 +44,8 @@
 	private int watches = -1;
 	private int connections = -1;
 
+	private boolean flag = false;
```
Comment (ruanwenjun) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/ZooKeeperState.java]: Can we use a different variable name, may be `healthFlag`?

File: dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/ZookeeperMonitor.java [hunk 3]
```diff
@@ -91,7 +89,7 @@
                 float avgLatency = state.getAvgLatency();
                 float maxLatency = state.getMaxLatency();
                 int nodeCount = state.getNodeCount();
-                int status = ok ? 1 : 0;
+                int status = state.isFlag() ? -1: (ok ? 1 : 0);
```
Comment (ruanwenjun) [inline dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/ZookeeperMonitor.java]: I think return true or false to front-end may be better than returning 0,1. 
And if return -1, what will the front-end show in ui?

**Comment by jon-qj** [comment]
Ok,I will correct the variable name.
I use three numbers to show the three different running states on the front-end.And actually,'ruok' does show the running state of node.
----- 原始邮件 -----
发件人：Wenjun Ruan ***@***.***>
收件人：apache/dolphinscheduler ***@***.***>
抄送人：jon-qj ***@***.***>,  Mention ***@***.***>
主题：Re: [apache/dolphinscheduler] [#5748][refactor]get more reasonable monitoring indexes of the zk cluster (#5766)
日期：2021年07月08日 13点53分


@ruanwenjun commented on this pull request.



In dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/FourLetterWordMain.java:
> @@ -25,10 +25,7 @@
 import java.io.IOException;
 import java.io.InputStreamReader;
 import java.io.OutputStream;
-import java.net.InetAddress;
-import java.net.InetSocketAddress;
-import java.net.Socket;
-import java.net.SocketTimeoutException;
+import java.net.*;

Don't use import *.

https://stackoverflow.com/questions/147454/why-is-using-a-wild-card-with-a-java-import-statement-bad



In dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/ZooKeeperState.java:
> @@ -43,6 +44,8 @@
 	private int watches = -1;
 	private int connections = -1;
 
+	private boolean flag = false;

Can we use a different variable name, may be healthFlag?



In dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/utils/ZookeeperMonitor.java:
> @@ -91,7 +89,7 @@
                 float avgLatency = state.getAvgLatency();
                 float maxLatency = state.getMaxLatency();
                 int nodeCount = state.getNodeCount();
-                int status = ok ? 1 : 0;
+                int status = state.isFlag() ? -1: (ok ? 1 : 0);

I think return true or false to front-end may be better than returning 0,1

&mdash;

**Comment by ruanwenjun** [comment]
@jon-qj Using number is ok, I just thought someone might be confused when he see -1, 0, 1. You don't need to change this, just my own opinion.
And you may need to change this pr commit to 1.3.7-prepare.

**Comment by ruanwenjun** [comment]
Back end LGTM.

**Comment by CalvinKirs** [comment]
close by #5768

**Pull request closed without merging**